packages/components/src/internal/components/samples/SampleDeleteMenuItem.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, memo, useCallback, useState } from 'react';
+import { MenuItem } from 'react-bootstrap';
+
+import { AuditBehaviorTypes } from '@labkey/api';
+
+import { EntityDeleteModal, SampleTypeDataType, QueryModel, SelectionMenuItem } from '../../..';
+
+interface Props {
+    queryModel: QueryModel;
+    key?: string;
+    itemText?: string;
+    verb?: string;
+    beforeSampleDelete?: () => any;
+    afterSampleDelete?: (rowsToKeep?: any[]) => any;
+    auditBehavior?: AuditBehaviorTypes;
+    maxDeleteRows?: number;
+}
+
+export const SampleDeleteMenuItem: FC<Props> = memo(props => {
+    const {
+        key,
+        itemText,
+        queryModel,
+        verb,
+        beforeSampleDelete,
+        afterSampleDelete,
+        auditBehavior,
+        maxDeleteRows,
+    } = props;
+    const [showConfirmDeleteSamples, setShowConfirmDeleteSamples] = useState<boolean>(false);
+
+    const onClick = useCallback(() => {
+        if (!queryModel || queryModel.selections.size > 0) {
+            setShowConfirmDeleteSamples(true);
+        }
+    }, [queryModel]);
+
+    const onClose = useCallback(() => {
+        setShowConfirmDeleteSamples(false);
+    }, []);
+
+    const onDeleteComplete = useCallback((rowsToKeep: any[]) => {
+        setShowConfirmDeleteSamples(false);
+        if (afterSampleDelete) afterSampleDelete(rowsToKeep);
+    }, []);
+
+    const useSelection = queryModel !== undefined;
```

Comment:
the name of this const seems odd. The right hand side doesn't have anything to do with selections. Should the const be removed and just put the `queryModel !== undefined` check inline below?

Reply:
I think it was named this way to distinguish cases when delete from a grid with selection, vs delete a single sample from detail page.